bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/gems/GemsResultsAnalyzer.scala

```diff
@@ -61,10 +61,12 @@ object GemsResultsAnalyzer {
     obsContext.getBaseCoordinates.asScalaOpt.map(_.toNewModel).map { base =>
       // gemsGuideStars needs to be mutable to support updates from inside mascot
       val gemsGuideStars = TiptiltFlexurePair.pairs(catalogSearch.asScala.toList).foldLeft(List.empty[GemsGuideStars]) { (gemsGuideStars, pair) =>
+        // tiptiltGroup and flexureGroup are each one of CanopusWfs.Group and GsaoiOdgw.group.
         val tiptiltGroup = pair.tiptiltResults.criterion.key.group
         val flexureGroup = pair.flexureResults.criterion.key.group
         val tiptiltTargetsList = filter(obsContext, pair.tiptiltResults.results, tiptiltGroup, posAngles.asScala.toSet)
         val flexureTargetsList = filter(obsContext, pair.flexureResults.results, flexureGroup, posAngles.asScala.toSet)
+        val tiptiltGroupPreAsterismFilter: (List[SiderealTarget] => Boolean) = lst => tiptiltGroup.asterismPreFilter(lst.asImList)
```

Comment:
What's the role of the `preFilter`

Reply:
`preFilter` is just the way to pass on the filtering to be done to asterism candidates to cut them out from consideration entirely before Strehl calculations are done. In the case of CWFS, this is to limit the diff by R-mag 3.